AcademyApp/ViewControllers/LoginViewController.swift

```diff
@@ -0,0 +1,189 @@
+//
+//  ViewController.swift
+//  AcademyApp
+//
+//  Created by Peter Sevcik on 13/06/2019.
+//  Copyright © 2019 Peter Sevcik. All rights reserved.
+//
+
+import UIKit
+
+class LoginViewController: UIViewController
+{
+    // MARK: - View Controller's life cycle methods
+    
+    override func viewDidLoad() {
+        super.viewDidLoad()
+        addObservers()
+        setupUI()
+        emailTextField.delegate = self
+        passwordTextField.delegate = self
+        
+        defaultDoneButtonBottomSpacing = doneButtonBottomConstraint.constant
+    }
+
+    deinit {
+        removeObservers()
+    }
+    
+    // MARK: - Stored Properities and methods
+    var defaultDoneButtonBottomSpacing : CGFloat!
+    
+    func isValidEmail(possibleEmail text: String)->Bool {
+        let emailRegEx = "[A-Z0-9a-z._%+-]+@[A-Za-z0-9.-]+\\.[A-Za-z]+"
+        let emailTest = NSPredicate(format:"SELF MATCHES %@", emailRegEx)
+        return emailTest.evaluate(with: text)
+    }
+    
+    // MARK: - @IBOutlets @IBActions
+    @IBOutlet weak var doneButtonBottomConstraint: NSLayoutConstraint!
+    @IBOutlet var horizontalLineViews: [UIView]!
```

Comment:
I didn't know this is even possible - #learn-something-new-every-day. Looks ugly with the exclamation mark though.